### PR TITLE
Treat a vector of wav paths as DistributedWAVFile

### PR DESCRIPTION
**Pull request by ymtoo**
The wav files can be from different folders.

### CODE REVIEW

File: src/LazyWAVFiles.jl [hunk 1]
```diff
@@ -82,6 +82,15 @@ struct DistributedWAVFile{T,N,L,FS} <: AbstractArray{T,N}
     blockarray::L
     fs::FS
 end
+function LazyWAVFiles.DistributedWAVFile(wavpaths::Vector{String})
```
Comment (baggepinnen) [inline src/LazyWAVFiles.jl]:
```suggestion
function DistributedWAVFile(wavpaths::Vector{String})
```

File: test/runtests.jl [hunk 2]
```diff
@@ -131,6 +131,16 @@ using Test, LazyWAVFiles, WAV, BenchmarkTools
...
+
+        d1 = mktempdir()
+        WAV.wavwrite(a, joinpath(d1,"fs800.wav"), Fs=800)
+        d2 = mktempdir()
+        WAV.wavwrite(b, joinpath(d2,"fs800.wav"), Fs=800)
+
+        wavpaths = vcat(joinpath.(d1, readdir(d1)), joinpath.(d2, readdir(d2)))
+         df = DistributedWAVFile(wavpaths)
+         @test df[1:10] == a
+         @test df[11:end] == b
```
Comment (baggepinnen) [inline test/runtests.jl]:
```suggestion
        df = DistributedWAVFile(wavpaths)
        @test df[1:10] == a
        @test df[11:end] == b
```

File: src/LazyWAVFiles.jl [hunk 3]
```diff
@@ -82,6 +82,15 @@ struct DistributedWAVFile{T,N,L,FS} <: AbstractArray{T,N}
...
+    isempty(wavpaths) && error("Found no wav paths.")
+    files = LazyWAVFile.(wavpaths)
+    fs0 = files[1].fs
+    if !all(x->x.fs==fs0, files)
+        error("WAV files in $folder have different sample rates.")
+    end
```
Comment (baggepinnen) [inline src/LazyWAVFiles.jl]:
```suggestion
    all(x->x.fs==fs0, files) || error("WAV files in $folder have different sample rates.")
```

**Review by baggepinnen — APPROVED** [review]
Nice idea :)

File: src/LazyWAVFiles.jl (hunk 3 above)
Comment (ymtoo) [reply src/LazyWAVFiles.jl]: Just found out a bug. Please remove `$folder` as it's undefined.

**Pull request merged**